universal/router/matches/WithSubRoutes.js

```diff
@@ -0,0 +1,13 @@
+/* @flow */
+import React from 'react'
+import { Match } from 'react-router'
+
+import type { RouteProps } from '../types'
+
+const WithSubRoutes = (route: RouteProps) => (
+  <Match {...route} render={(props) => (
+    <route.component {...props} store={route.store} routes={route.routes} />
+  )} />
+)
+
+export default WithSubRoutes
```

Comment:
I struggle with this pattern here, I'd prefer to stay with their API rather than off-the-bat creating our own abstractions.